### PR TITLE
Adds gene_symbols option to scanpy rank_genes plots and logs to scatterplot

### PR DESCRIPTION
**Pull request by pcm32**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Adds the gene symbol option to all the rank_genes plots.

### CODE REVIEW

**Comment by pcm32** [comment]
All linting errors seem to be from outdated URLs or part of the tool that were not modified in this PR, are we allowed to merge with those?

**Comment by bgruening** [comment]
It would be better for the users to fix those I think.

**Comment by pcm32** [comment]
Current conda mulled install as it was would not work anymore as it install matplotlib at version 3.7.x, which breaks Scanpy, so, so I aimed to fix some relevant dependencies to be the same as in the biocontainers `quay.io/biocontainers/scanpy:1.7.1--py_0` although perhaps I should have looked at the mulled container `mulled-v1-613317b83503dd6dc670a806c0074cb4b966be6eb10b77d40f34e80ba42afafe`.

**Comment by pcm32** [comment]
I have fixed all URLs I think, but not sure what to do with https://github.com/galaxyproject/tools-iuc/actions/runs/4518695863/jobs/7958748405#step:5:582 those errors.

**Comment by bgruening** [comment]
@pcm32 the delta for the images looks very large. Can you regenerate the images and see if they still mak sense?

**Comment by pcm32** [comment]
Is there an easy way to get the outputs from planemo?

**Comment by bgruening** [comment]
Easy, but use with care, is: `planemo test --update_test_data`

**Comment by pcm32** [comment]
All size varying PNGs looked the same as previous ones. I have replaced with the newer ones and removed deltas.

**Comment by bgruening** [comment]
I restarted all tests.

**Comment by pcm32** [comment]
Thanks! This seems again to be in some stale state.

**Comment by pcm32** [comment]
I think that this might require another tyre kicking please @bgruening ? By the way, how do I talk to you on Matrix / gitter? I keep sending messages to a chat with you, but maybe you are not getting them? I went in today and also had messages from people weeks ago for which I didn't get notified, so maybe that is what is happening.

**Comment by bgruening** [comment]
Can you rebase, it seems a restart does not change it.

My matrix ID is: @bgruening:matrix.org

**Comment by pcm32** [comment]
@bgruening tools that haven't been modified are coming with this unspecific error now:

```
scanpy_inspect (Test #9)

Input staging problem: ("Connection broken: InvalidChunkLength(got length b'', 0 bytes read)", InvalidChunkLength(got length b'', 0 bytes read))
```

maybe this needs a re-run?

File: tools/scanpy/macros.xml [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
     <token name="@version@">1.7.1</token>
     <token name="@profile@">19.01</token>
-    <token name="@galaxy_version@"><![CDATA[@version@+galaxy0]]></token>
+    <token name="@galaxy_version@"><![CDATA[@version@+galaxy1]]></token>
     <xml name="requirements">
         <requirements>
             <requirement type="package" version="@version@">scanpy</requirement>
+            <requirement type="package" version="3.6.3">matplotlib</requirement>
+            <requirement type="package" version="0.7.5">anndata</requirement>
             <requirement type="package" version="2.0.17">loompy</requirement>
```
Comment (bgruening) [inline tools/scanpy/macros.xml:10]: can it be that you need to bump loompy and other deps as well to keep them compatible?

Comment (pcm32) [reply tools/scanpy/macros.xml:10]: I pinned those to the version on the current mulled container used for that tool I think, to prevent issues.

Comment (pcm32) [reply tools/scanpy/macros.xml:10]: Also, locally all tests pass (at least for plot and inspect (the one that failed in the last actions run). I suspect it was a networking issue on the last github actions run :-( .

**Comment by pcm32** [comment]
Errored jobs now don't show any logs to see what is the problem now....

**Comment by pcm32** [comment]
Added log option for scatter (when x and y come from obs or var).

**Comment by pcm32** [comment]
What should we do with this @bgruening ? Thanks!

**Comment by pcm32** [comment]
If you look at the report for the other tools, they are all passing.

**Comment by pcm32** [comment]
It's very strange that it is now the lint that fails, which had never failed before.

**Comment by bgruening** [comment]
That part is less strange. We get stricter with linting over time and add new features as we go to the lining. What makes me nervous is that this one test is still running. I guess one of the tools has problems to resolve the conda environment?

**Comment by pcm32** [comment]
...mmmm... as far as I remember, all tools use the same environment. Isn't this also pulling the container you prepared?

**Comment by pcm32** [comment]
The planemo test for part 0 shows up as in an "unknown" state, I don't remember seeing that often. Very strange.

**Comment by bernt-matthias** [comment]
In the CI summary you can see the reason:

```
The job running on runner GitHub Actions 7 has exceeded the maximum execution time of 360 minutes.
```

**Comment by pcm32** [comment]
Lets see if we can rescue this once and for all @bgruening :-).

**Comment by bgruening** [comment]
@pavanvidem can you maybe take a look here and take the changes over?

**Comment by pavanvidem** [comment]
All the changes in this PR were included in the latest version 1.9.6 of scanpy. Here is the link to the commit: https://github.com/pavanvidem/tools-iuc/commit/30f1d1787b1aa8f2896f5e9cd328260d4a9605e8

This PR can be closed if @pcm32 agrees.

**Comment by bgruening** [comment]
Cool, thanks Pavan. I will close, Pablo let me know if there is still something missing.

**Pull request closed without merging**